tools/syntax_gen.py

```diff
@@ -118,7 +118,7 @@ def generate(outf, name, tags, members, alltypes):
 	outf.write('        {}({}{}){}\n'.format(base, kindValue, baseInitializers, initializers))
 	outf.write('    {\n')
 
-	if len(members) == 0:
+	if len(members) == 0 and not forceEmpty:
```

Comment:
I think instead of adding 'nothing' you could just make a small tweak and reuse empty. Right now everything that uses 'empty' also sets final=false. You can just check here if final=True (which is the default) and then output the functions in that case.